frontend/src/components/object/ObjectFileDetails.vue

```diff
@@ -74,6 +75,18 @@ function onDeletedSuccess() {
   router.push({ name: RouteNames.LIST_OBJECTS, query: { bucketId: bucketId.value } });
 }
 
+async function onFileUploaded() {
+  await objectStore.fetchObjects({objectId: props.objectId, userId: getUserId.value, bucketPerms: true});
+  await versionStore.fetchVersions({ objectId: props.objectId });
+
+  // Obtaining the version id and passing it to the route forces a destruct/construct of the component
+  // Easier approach than attempting to in-place refresh all the data
+  router.push({ name: RouteNames.DETAIL_OBJECTS, query: {
+    objectId: props.objectId,
+    versionId: versionStore.findLatestVersionIdByObjectId(props.objectId)
+  }});
+}
```

Comment:
I can't quite remember, but there may be a way to only update a certain set of sub-components by force updating its id key and then having that component re-render. What you have here should work, and also properly addresses the url location though.